ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -107,6 +107,15 @@ def register_measurement_service(
             service_descriptor.description_url = service_info.description_url
             service_descriptor.provided_interfaces.extend(service_info.provided_interfaces)
 
+            service_info_annotations = service_info.annotations
+            for keys in service_info_annotations:
+                if isinstance(service_info_annotations[keys], (int, float)):
+                    service_descriptor.annotations[keys] = str(service_info_annotations[keys])
+                else:
+                    service_descriptor.annotations[keys] = " ".join(
```

Comment:
I don't think this will handle the possibility of complex types that the user could provide.  A dictionary that contains a dictionary that contains an array of strings will need to be converted to embedded JSON.  Simply joining with a space won't do that, and embedded JSON usually removes all whitespace outside of a string.